src/atomicdex/models/qt.orders.model.cpp

```diff
@@ -645,6 +610,48 @@ namespace atomic_dex
         this->m_model_data.clear();
         this->endResetModel();
     }
+    
+    std::pair<std::string, std::string> orders_model::determine_amounts_in_current_currency(const std::string& base_coin, const std::string& base_amount,
+                                                                      const std::string& rel_coin, const std::string& rel_amount) noexcept
+    {
+        const auto&     settings_system     = m_system_manager.get_system<settings_page>();
+        const auto&     mm2                 = m_system_manager.get_system<mm2_service>();
+        const auto&     base_coin_info      = mm2.get_coin_info(base_coin);
+        const auto&     rel_coin_info       = mm2.get_coin_info(rel_coin);
+        const auto&     global_price_system = m_system_manager.get_system<global_price_service>();
+        const auto&     current_fiat        = settings_system.get_current_fiat().toStdString();
+        std::string     base_amount_fiat    = "0";
+        std::string     rel_amount_fiat     = "0";
+        std::error_code ec;
+
+        if (base_coin_info.coinpaprika_id != "test-coin")
+        {
+            base_amount_fiat = global_price_system.get_price_as_currency_from_amount(current_fiat, base_coin, base_amount, ec);
+        }
+        if (rel_coin_info.coinpaprika_id != "test-coin")
+        {
+            rel_amount_fiat = global_price_system.get_price_as_currency_from_amount(current_fiat, rel_coin, rel_amount, ec);
+        }
+        return std::make_pair(base_amount_fiat, rel_amount_fiat);
+    }
+    
+    std::pair<std::string, std::string> orders_model::determine_amounts_in_current_currency(const ::mm2::api::swap_contents& contents)
+    {
+        bool is_maker = boost::algorithm::to_lower_copy(contents.type) == "maker";
+        
+        if (is_maker)
+        {
+            return determine_amounts_in_current_currency(contents.maker_coin, contents.maker_amount, contents.taker_coin, contents.taker_amount);
+        }
+        return determine_amounts_in_current_currency(contents.taker_coin, contents.taker_amount, contents.maker_coin, contents.maker_amount);
+    }
+    
+    void orders_model::on_current_currency_changed(const current_currency_changed&) noexcept
+    {
+        spdlog::trace("updated orders model");
+        refresh_or_insert_orders();
```

Comment:
I think here it's better to retrieve mm2 system and call the batch that refresh orders and swap since it's handling different internals for you